components/integrations/bots/index.js

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
+import {getBotAccounts} from 'mattermost-redux/selectors/entities/bots';
+import {loadBots, disableBot, enableBot} from 'mattermost-redux/actions/bots';
+import {createUserAccessToken, revokeUserAccessToken, enableUserAccessToken, disableUserAccessToken, getUserAccessTokensForUser, getUser} from 'mattermost-redux/actions/users';
+import * as UserSelectors from 'mattermost-redux/selectors/entities/users';
+
+import Bots from './bots.jsx';
+
+function mapStateToProps(state) {
+    const bots = getBotAccounts(state);
+    const owners = Object.values(bots).
```

Comment:
This is returning a new object each time. Could we instead have the child `Bot` component be in charge of getting the owner from redux? It doesn't look like `this.props.owners` is used anywhere else in the `Bots` component

Reply:
The bot component is not connected to redux. I could connect it just for this. Not sure what would be more performant though. I could also use some derived state... not really sure about any of those options.